integrations/mongodb/documentation.yaml

```diff
@@ -0,0 +1,90 @@
+exporter_type: standalone
+app_name_short: MongoDB
+app_name: {{app_name_short}}
+app_site_name: {{app_name_short}}
+app_site_url: https://www.mongodb.com/
+exporter_pkg_name: mongodb_exporter
+exporter_repo_url: https://github.com/percona/mongodb_exporter
+dashboard_available: true
+minimum_exporter_version: "0.20"
+multiple_dashboards: false
+dashboard_display_name: {{app_name_short}} Prometheus Overview
+standalone_config: |
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: mongodb-exporter
+    labels:
+      app.kubernetes.io/name: mongodb-exporter
+  spec:
+    replicas: 1
+    selector:
+      matchLabels:
+        app.kubernetes.io/name: mongodb-exporter
+    template:
+      metadata:
+        labels:
+          app.kubernetes.io/name: mongodb-exporter
+      spec:
+        containers:
+        - name: exporter
+          image: percona/mongodb_exporter:0.20 
+          args:
+          - --mongodb.uri=mongodb://my-user:password@mongodb-client:27017
+          ports:
+          - name: prometheus
+            containerPort: 9216
+            protocol: TCP
+additional_install_info: |
+  The {{app_name_short}} exporter is configurable with [flags]({{exporter_repo_url}}#configuration){:class=external},
+  which can be set as container args. You can adjust your exporter's configuration to suit your {{app_name_short}} configuration.
+  Update the `--mongodb.uri` flag to match your environment's username, password, and clusterIP service name.
+podmonitoring_config: |
+  apiVersion: monitoring.googleapis.com/v1
+  kind: PodMonitoring
+  metadata:
+    name: mongodb
+    labels:
+      app.kubernetes.io/name: mongodb
+      app.kubernetes.io/part-of: google-cloud-managed-prometheus
+  spec:
+    endpoints:
+    - port: prometheus
+      scheme: http
+      interval: 30s
+      path: /metrics
+    selector:
+      matchLabels:
+        app.kubernetes.io/name: mongodb-exporter
+additional_prereq_info: |
+  The {{app_name_short}} exporter is configurable with [flags](https://hub.docker.com/r/percona/mongodb_exporter){:class=external},
+  which can be set as container args. You can adjust your exporter's configuration to suit your {{app_name_short}} configuration.
+
+  Update the `--mongodb.uri` flag to match your environment's username, password, and clusterIP service.
```

Comment:
This seems identical to additional_install_info? Let's omit lines 59 - 63 and just use additional_install_info. 

The primary difference between the twois that in the template this would render before standalone_config, while additional_install_info would render after standalone_config. In this case, the latter seems more appropriate.

Reply:
I removed the `additional_prereq_info` section. I added a comment around the exporter not being compatible with MongoDB version 6.0.4. When I had that version running, the exporter only returned `mongodb_up`. I'm uncertain if there's something I'm missing where configuration is concerned.

I read this thread where it seems the exporter was able to work w/ version `5.0.6`, which is why I believe I may have misconfigured something. My steps, deploying using the community operator:
1. Updated the operator image tags and the version of MongoDB in the yaml
2. Left the `prometheus:` section out, so as to not setup and endpoint with auth
3. Setup the exporter the same way we had it setup for `4.2.6`

The result is that the exporter connects and returns some metrics, including `mongodb_up`, but there are no other metrics with a `mongodb` prefix.

Reply:
Thanks for flagging this! I've approved this for now so we can publish it and get value out of it. Let me look into the mongodb v6 issue on my end as well next week.

Reply:
I learned why I was only seeing the `mongodb_up` metric coming back from the exporter and that the community operator for MongoDB allows us to configure an endpoint that [exposes prometheus metrics](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/config/samples/mongodb.com_v1_mongodbcommunity_prometheus.yaml).

I misconfigured the exporter, initially, needed to pass the `--collect-all` argument, which was not needed prior.
```yaml
  args:
  - --mongodb.uri=mongodb://my-user:password@example-mongodb-svc:27017
  - --discovering-mode
  - --collect-all
```

After that, (testing v6.0.5 of MongoDB) the dashboard is working as expected. The endpoint that the operator allows us to configure forces us to require basic authentication. The PodMonitoring resource can't be setup with authentication, to my knowledge.